src/planner/binder/query_node/bind_select_node.cpp

```diff
@@ -445,6 +445,7 @@ unique_ptr<BoundQueryNode> Binder::BindNode(SelectNode &statement) {
 	vector<LogicalType> internal_sql_types;
 	for (idx_t i = 0; i < statement.select_list.size(); i++) {
 		bool is_window = statement.select_list[i]->IsWindow();
+		bool is_window_aggregate = statement.select_list[i]->type == ExpressionType::WINDOW_AGGREGATE;
```

Comment:
This checks that the window function is specifically a window aggregate - it misses other window functions (such as `WINDOW_LEAD`, e.g. `lead(r) over ()`):

```sql
select unnest([5,4,3]) as r, lead(r) OVER ();
```

Could you add a test for that as well to catch this case?

Reply:
This also only checks that the root element is a window clause, what if we have something like this:

```sql
select unnest([5,4,3]) as r, sum(r) OVER () + 42;
```